pynio/service.py

```diff
@@ -98,6 +102,12 @@ def command(self, command, block=None, **request_kwargs):
                                                   command),
                        **request_kwargs)
 
+    def create_block(self, name, type, config=None):
+        blk = Block(name, type, config=config, instance=self._instance)
+        self.connect(blk)
+        if self._instance:
+            self.save()
```

Comment:
save should be called on the new block too. Otherwise this service isn't going to work! 

Have this method return the new block. That way the user can at least add the block to the instance manually if the Service does not have one here.